pkg/listener/mysql.go

```diff
@@ -520,47 +515,44 @@ func scramblePassword(scramble []byte, password string) []byte {
 }
 
 func (l *MysqlListener) ExecuteCommand(ctx context.Context, c *mysql.Conn, data []byte) error {
-	newCtx, span := tracing.GetTraceSpan(ctx, "mysql_exec_command")
-	defer span.End()
-
 	commandType := data[0]
 	switch commandType {
 	case constant.ComQuit:
 		// https://dev.constant.Com/doc/internals/en/com-quit.html
 		c.RecycleReadPacket()
-		connectionID := proto.ConnectionID(newCtx)
-		l.executor.ConnectionClose(proto.WithConnectionID(newCtx, connectionID))
+		connectionID := proto.ConnectionID(ctx)
+		l.executor.ConnectionClose(proto.WithConnectionID(ctx, connectionID))
 		log.Debugf("connection closed, id: %d", connectionID)
 		return errors.New("ComQuit")
 	case constant.ComInitDB:
 		db := string(data[1:])
 		c.RecycleReadPacket()
 		l.schemaName = db
-		err := l.executor.ExecuteUseDB(newCtx, db)
+		err := l.executor.ExecuteUseDB(ctx, db)
 		if err != nil {
-			tracing.RecordErrorSpan(span, err)
 			return err
 		}
 		if err := c.WriteOKPacket(0, 0, c.StatusFlags(), 0); err != nil {
 			log.Errorf("Error writing ComInitDB result to %s: %v", c, err)
-			tracing.RecordErrorSpan(span, err)
 			return err
 		}
 	case constant.ComQuery:
 		err := func() error {
 			c.StartWriterBuffering()
+			query := string(data[1:])
+			c.RecycleReadPacket()
+			p := parser.New()
+			stmt, err := p.ParseOneStmt(query, "", "")
```

Comment:
There seems to be something wrong with this code, err judgment should be on the next line. stmt may be nil if an exception occurs.